src/check_update.ts

```diff
@@ -82,43 +82,102 @@ async function createVersionMap() {
   return versionMap;
 }
 
-/** readProjectDependencies reads from possible dependency files
- * (import_map.json, slack.json) and maps them to the versionMap
- * containing each dependency's update information
+/** readProjectDependencies cycles through supported project
+ * dependency files and maps them to the versionMap that contains
+ * each dependency's update information.
  */
-async function readProjectDependencies(): Promise<VersionMap> {
+export async function readProjectDependencies(): Promise<VersionMap> {
   const cwd = Deno.cwd();
   const versionMap: VersionMap = {};
+  const dependencyFiles = await gatherDependencyFiles(cwd);
 
-  // Find SDK component versions in import map, if available
-  const map = await getJSON(`${cwd}/import_map.json`);
-  for (const sdkUrl of Object.values(map.imports) as string[]) {
-    for (const sdk of IMPORT_MAP_SDKS) {
-      if (sdkUrl.includes(sdk)) {
-        versionMap[sdk] = {
-          name: sdk,
-          current: extractVersion(sdkUrl),
-        };
+  for (const [file, depKey] of dependencyFiles) {
+    const fileJSON = await getJSON(`${cwd}/${file}`);
+    const fileDependencies = extractDependencies(fileJSON, depKey);
+
+    // For each dependency found, compare to SDK-related dependency
+    // list and, if known, update the versionMap with version information
+    for (const [_, val] of fileDependencies) {
+      for (const sdk of [...IMPORT_MAP_SDKS, ...SLACK_JSON_SDKS]) {
+        if (val.includes(sdk)) {
+          versionMap[sdk] = {
+            name: sdk,
+            current: extractVersion(val),
+          };
+        }
       }
     }
   }
 
-  // Find SDK component versions in slack.json, if available
-  const { hooks }: { [key: string]: string } = await getJSON(
-    `${cwd}/slack.json`,
-  );
-  for (const command of Object.values(hooks)) {
-    for (const sdk of SLACK_JSON_SDKS) {
-      if (command.includes(sdk)) {
-        versionMap[sdk] = {
-          name: sdk,
-          current: extractVersion(command),
-        };
-      }
+  return versionMap;
+}
+
+/**
+ * gatherDependencyFiles rounds up all SDK-supported dependency files, as well
+ * as those dependency files referenced in deno.json or deno.jsonc, and returns
+ * an array of arrays made up of filename and dependency key pairs.
+ */
+export async function gatherDependencyFiles(cwd: string): Promise<string[][]> {
+  const dependencyFiles = [
+    ["import_map.json", "imports"],
+    ["import_map.jsonc", "imports"],
```

Comment:
I think we could remove this one. Deno [decided to not support jsonc files](https://github.com/denoland/deno/issues/12993) for import maps because it would be incompatible with the web standard.

Reply:
Less files to keep track of, the better! But, if we ever want to reintroduce it, it'll work at least.

Related question: `slack.jsonc` doesn't technically work, I don't think. When I tested it out, something else fails -- I assume on the CLI side of things. Do we want to remove that, as well, assuming that's where the failure is occurring?

Reply:
I'm seeing the same, where if I use a `slack.jsonc` file I just get `✔ You are using the latest Slack CLI and SDK`. IMO it'd be great if we support it in the hook now so whenever the CLI adds support it will just work.